### ISSUE TITLE
Inconsistency in base cnt and call cnt when switching 2 VCFs as base and call files in truvari bench

### ISSUE THREAD

**Issue by nmtrang00** [ACEnglish/truvari#102]
**Version** : 

Truvari v3.1.0

**Describe the bug** : 

I used 2 VCFs (tool1_out.vcf.gz and tool2_out.vcf.gz) as the input for **truvari bench**.
The commands are as followed:
"""
#Tool 1 as truthset
truvari bench -b tool1_out.vcf.gz \
    -c tool2_out.vcf.gz \
    -f $fasta \
    --pctsize 0 \
    --refdist 1000 \
    -o ${outdir}/tool1_as_truthset

#Tool 2 as truthset
        truvari bench -b  tool2_out.vcf.gz \
        -c tool1_out.vcf.gz \
        -f $fasta \
        --pctsize 0 \
        --refdist 1000 \
        -o ${outdir}/tool2_as_truthset

summary_1=${outdir}/tool1_as_truthset/summary.txt
summary_2=${outdir}/tool2_as_truthset/summary.txt
 
"""
I expected the "base cnt" from **summary_1** to be equal with the "call cnt" from **summary_2**, and vice versa.
However, the number of "base cnt" from **summary_1** was 1 call fewer than the "call cnt" from **summary_2**.
I have tried with 24 couples of VCFs and only 10 out of them had this problem.

**Comment by ACEnglish** [comment]
Hello,

Unfortunately I'm unable to recreate this error.
```bash
python3 -m pip install truvari==3.1.0
call1=call_one.vcf.gz
call2=call_two.vcf.gz
ref=GRCh38_1kg_mainchrs.fa

truvari bench -b $call1 -c $call2  -f $ref --pctsize 0 --refdist 1000 -o one_on_two
truvari bench -b $call2 -c $call1  -f $ref --pctsize 0 --refdist 1000 -o two_on_one
```

one_on_two summary:
```json
{
    "TP-base": 4460,
    "TP-call": 4460,
    "FP": 5750,
    "FN": 4176,
    "precision": 0.43682664054848186,
    "recall": 0.5164427975914775,
    "f1": 0.47330998620396897,
    "base cnt": 8636,
    "call cnt": 10210,
    "TP-call_TP-gt": 3026,
    "TP-call_FP-gt": 1434,
    "TP-base_TP-gt": 3026,
    "TP-base_FP-gt": 1434,
    "gt_concordance": 0.6784753363228699
}
```

two_on_one summary:
```json
{
    "TP-base": 4460,
    "TP-call": 4460,
    "FP": 4176,
    "FN": 5750,
    "precision": 0.5164427975914775,
    "recall": 0.43682664054848186,
    "f1": 0.47330998620396897,
    "base cnt": 10210,
    "call cnt": 8636,
    "TP-call_TP-gt": 3026,
    "TP-call_FP-gt": 1434,
    "TP-base_TP-gt": 3026,
    "TP-base_FP-gt": 1434,
    "gt_concordance": 0.6784753363228699
}
```

Are you sure there were no other parameters (e.g. `--includebed`) and that the VCFs have consistent header `##contig` definitions? Those are the only things I can think of.

**Comment by nmtrang00** [comment]
Hi,
Thanks for your reply.
I attached the VCFs to this email. Please have a look at them.
Thanks,
Trang

**Comment by ACEnglish** [comment]
There's nothing attached. Github will screen email attachments with 'suspect' file extensions

**Comment by nmtrang00** [comment]
I attached those files to the reply email, but they did not appear here. Please check those file I attached here.
Thanks,
Trang
[tool1.vcf.gz](https://github.com/ACEnglish/truvari/files/8251769/tool1.vcf.gz)
[tool2.vcf.gz](https://github.com/ACEnglish/truvari/files/8251770/tool2.vcf.gz)

**Comment by ACEnglish** [comment]
This isn't a problem. It can be replicated and explained.
tl;dr - See [Wiki](https://github.com/ACEnglish/truvari/wiki/bench#difference-between---sizemin-and---sizefilt) for details

```bash
call1=tool1.vcf.gz
call2=tool2.vcf.gz
ref=/users/u233287/scratch/insertion_ref/msru/data/reference/grch38/GRCh38_1kg_mainchrs.fa

rm -rf one_on_two
rm -rf two_on_one

truvari bench -b $call1 -c $call2  -f $ref --pctsize 0 --refdist 1000 -o one_on_two  --debug
truvari bench -b $call2 -c $call1  -f $ref --pctsize 0 --refdist 1000 -o two_on_one  --debug

vimdiff <(grep -v "#" one_on_two/tp-base.vcf | cut -f 1,2 | sort) <(grep -v "#" two_on_one/tp-call.vcf | cut -f 1,2 | sort)
# Found the difference in two_on_one/tp-call.vcf
# chr13   32912300

bcftools query -f "%CHROM\t%POS\t%SVLEN\t%MatchId\n" two_on_one/tp-call.vcf | grep -w 32912300
# chr13   32912300        46      21.0.0

# This call is smaller than --sizemin but larger than --sizefilt
# It matches to 21.0.0
bcftools query -f "%CHROM\t%POS\tID\n"  two_on_one/tp-base.vcf -i "MatchId=='21.0.0'"
# chr13   32912300        rs886038095-rs397507695

# and that call in one_on_two/ is a false positive
grep -w rs886038095-rs397507695 one_on_two/*.vcf
# one_on_two/fp.vcf:chr13 32912300        rs886038095-rs397507695
```

So the variant being shorter than than `--sizemin` is why the counts are different. If you'd like them to be consistent, use 
`--sizemin 0 -sizefilt 0`
Note, I think I used 3.2.0-dev. But I'm pretty confident that the same thing is happening in the current release.

**Comment by ACEnglish** [comment]
Note: I reworded the wiki entry a little to add more clarity

**Comment by nmtrang00** [comment]
I got it. Thanks for your help.
